File: ECMAIN/fix_tunnel.py
from pbxproj import XcodeProject
try:
    project = XcodeProject.load('/Users/hh/Desktop/my/ECMAIN/ECMAIN.xcodeproj/project.pbxproj')
    target = project.get_target_by_name('Tunnel')
    if target:
        files = project.get_build_files_for_target(target.name)
        for f in files:
            file_ref = project.objects.get(f.fileRef)
            if file_ref and hasattr(file_ref, 'name') and file_ref.name and file_ref.name in ['ECTaskPollManager.m', 'ECTaskListViewController.m']:
                project.remove_build_file(f.get_id())
                print('Removed', file_ref.name)
            elif file_ref and hasattr(file_ref, 'path') and file_ref.path and ('ECTaskPollManager.m' in file_ref.path or 'ECTaskListViewController.m' in file_ref.path):
                project.remove_build_file(f.get_id())
                print('Removed path', file_ref.path)
        project.save()
        print('Fixed pbxproj.')
except Exception as e:
    print('Failed:', e)

File: ECMAIN/fix_tunnel.rb
require 'xcodeproj'
project_path = '/Users/hh/Desktop/my/ECMAIN/ECMAIN.xcodeproj'
project = Xcodeproj::Project.open(project_path)

tunnel_target = project.targets.find { |t| t.name == 'Tunnel' }
if tunnel_target
  files_to_remove = []
  tunnel_target.source_build_phase.files.each do |build_file|
    file_ref = build_file.file_ref
    if file_ref && file_ref.respond_to?(:path) && file_ref.path
      if file_ref.path.include?('ECTaskPollManager.m') || file_ref.path.include?('ECTaskListViewController.m')
        files_to_remove << build_file
        puts "Found #{file_ref.path} in Tunnel target"
      end
    end
  end
  
  files_to_remove.each do |bf|
    tunnel_target.source_build_phase.remove_build_file(bf)
    puts "Removed!"
  end
  
  project.save
  puts "Successfully cleaned Tunnel target."
else
  puts "Tunnel target not found."
end
